ft-sdk/src/layout.rs

```diff
@@ -46,11 +46,11 @@ pub trait Layout {
         Self: Sized,
     {
         let in_ = ft_sdk::In::from_request(r)?;
-        let mut l = Self::from_in(in_, RequestType::Page)?;
+        let mut l = Self::from_in(in_.clone(), RequestType::Page)?;
```

Comment:
Instead of cloning `in_`, you only need the `.set_cookies`, so clone just that bit, and pass just that to `ft_sdk::json_response()`.